### PR TITLE
Add data-group support to update dataset logic 

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**

Add test flow `uat1/_ _Lumen tests/Lumen Testing/2 Lumen RQG Testing/New form`

```clojure
(tu/read-edn-flow-dataset "uat1" "638889127" "638879132")
```

Screenshot 2021-01-07 at 12 54 14

### CODE REVIEW

File: backend/src/akvo/lumen/lib/transformation/engine.clj [hunk 1]
```diff
@@ -163,18 +163,21 @@
     (throw (ex-info "Failed to transform. Transformation can't have columns from more than one data-group specified"
                     {:transformation transformation}))))
 
+(defn data-group-by-tx [transformation data-groups]
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/engine.clj:166]: rename to data-group-by-transformation

File: backend/src/akvo/lumen/lib/transformation/engine.clj [hunk 2]
```diff
@@ -191,7 +194,6 @@
       (throw (ex-info (or message "") {:transformation-result (select-keys res [:success? :message :columns :execution-log])})))
     {:data-groups-to-be-created (or data-groups-to-be-created [])
      :columns columns
-     :previous-columns previous-columns
      :data-group data-group}))
 
 (defn adapt [data-groups-to-be-created data-groups]
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/engine.clj:199]: rename function

**Review by kardan — COMMENTED** [review]
Changes looks fine to me. But, while running all project test via cider I got errors with the test-data-groups-transformations test. Not sure if this was related or whatnot. 


akvo.lumen.lib.transformation-test
1 non-passing tests:

Fail in test-data-groups-transformations
Valid log

expected: ({:rnum 1, :c1 "A   ", :c2 "b   ", :c3 "c"})          

  actual: ({:rnum 1, :c1 "a   ", :c2 "b   ", :c3 "c"})          
    diff: - [{:c1 "A   "}]          
          + [{:c1 "a   "}]

**Comment by tangrammer** [comment]
> Changes looks fine to me. But, while running all project test via cider I got errors with the test-data-groups-transformations test. Not sure if this was related or whatnot.
> 
> akvo.lumen.lib.transformation-test
> 1 non-passing tests:
> 
> Fail in test-data-groups-transformations
> Valid log
> 
> expected: ({:rnum 1, :c1 "A ", :c2 "b ", :c3 "c"})
> 
> actual: ({:rnum 1, :c1 "a ", :c2 "b ", :c3 "c"})
> diff: - [{:c1 "A "}]
> + [{:c1 "a "}]

@kardan can you try again pulling last commit https://github.com/akvo/akvo-lumen/pull/3084/commits/b28e9026a0125ca278eddbf8297e223551f9fe7e ?

**Comment by kardan** [comment]
Tested again and all worked fine.

**Review by kardan — APPROVED** [review]
LGTM

**Pull request merged**